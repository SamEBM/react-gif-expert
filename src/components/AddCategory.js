import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = (props) => {
    const [inputValue, setInputValue] = useState('');
    const handleInput = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.length > 2) {
            props.addCategory(inputValue);
            setInputValue('');
        }
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input type="text" value={inputValue} onChange={handleInput}></input>
            </form>
        </>
    )
}

AddCategory.propTypes = {
    addCategory: PropTypes.func.isRequired
}

export default AddCategory