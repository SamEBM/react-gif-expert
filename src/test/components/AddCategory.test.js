import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas con Add Category', () => { 
    const addCategory = jest.fn();
    let wrapper = shallow(<AddCategory addCategory={addCategory}/>);

    // Esto se ejecuta cada vez que se ejecuta un test
    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory addCategory={addCategory}/>);
    });

    test('Debe mostrar <AddCategory /> correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    // Simular cambios en un Input
    test('Debe de cambiar el input', () => { 
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: {value} });
    })

    test('No debe hacer submit', () => { 
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(addCategory).not.toHaveBeenCalled();
    })

    test('Debe llamar addCategory y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        // Simula el inputChange
        input.simulate('change', { target: {value} }); 

        // Simula el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(addCategory).toHaveBeenCalled();
        expect(input.prop('value')).toBe('');
    })
})