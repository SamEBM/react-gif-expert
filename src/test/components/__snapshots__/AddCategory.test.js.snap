// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con Add Category Debe mostrar <AddCategory /> correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
